### PR TITLE
Support GNOME 3.32

### PR DESCRIPTION
**Pull request by jonafato**
This change updates the use of JavaScript classes to be compatible with
ES6 and GNOME Shell 3.32's behavior. Additionally, it adds a hack to
prevent calling `destroy` on an already destroyed indicator instance.

### CODE REVIEW

File: icon-hider@kalnitsky.org/_config.js [hunk 1]
```diff
@@ -11,10 +11,10 @@
 
 // extension name
 // used as identifier when adding to status area
-const EXTENSION_NAME = 'Icon Hider';
+var EXTENSION_NAME = 'Icon Hider';
```
Comment (ikalnytskyi) [inline icon-hider@kalnitsky.org/_config.js]: Why this change? What's wrong with `const`?

Comment (jonafato) [reply icon-hider@kalnitsky.org/_config.js]: GNOME logs warnings when using `const`
>Any symbols to be exported from a module must be defined with 'var'. The property access will work as previously for the time being, but please fix your code anyway.

This change isn't strictly necessary, and I can undo it if you object. I was mostly trying to get quieter logs when debugging. Per [this stackoverflow answer](https://stackoverflow.com/a/32558929), imports cannot be reassigned by the code that imports them, so the behavior here should still be consistent.

**Comment by jonafato** [comment]
I've updated my changeset to be much smaller here. There's a hack included to address #39. I'm not crazy about it, and I'm definitely open to better ideas here, but nothing is broken as far as I can tell. Feedback appreciated.

File: icon-hider@kalnitsky.org/metadata.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-    "shell-version": ["3.26", "3.28", "3.30"],
+    "shell-version": ["3.26", "3.28", "3.30", "3.32"],
```
Comment (theychx) [inline icon-hider@kalnitsky.org/metadata.json]: Your new revised extension does not work with gs <3.32, so the metadata should reflect that. We can have both the new and old version active on extensions.gnome.org, so that people get the one that works with their particular gs.

**Comment by theychx** [comment]
@jonafato Works great, thank you!
@ikalnytskyi Any thoughts?

File: icon-hider@kalnitsky.org/metadata.json (hunk 2 above)
Comment (jonafato) [reply icon-hider@kalnitsky.org/metadata.json]: Thanks, I've pushed an updated version declaring support for only 3.32. For future reference, is there a good method of checking compatibility (using TravisCI or something) of different GNOME versions? I have a feeling that a lot of extensions extend their lists of supported versions without pruning them.
